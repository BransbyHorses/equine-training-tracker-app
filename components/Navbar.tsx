import * as React from 'react';
import AppBar from '@mui/material/AppBar';
import Box from '@mui/material/Box';
import Toolbar from '@mui/material/Toolbar';
import Typography from '@mui/material/Typography';
import Button from '@mui/material/Button';
import IconButton from '@mui/material/IconButton';
import MenuIcon from '@mui/icons-material/Menu';
import { signIn, signOut, useSession } from 'next-auth/react';
import HomeIcon from './HomeIcon';

export default function ButtonAppBar() {
    const { data: session, status } = useSession();
    const loading = status === 'loading';

    const signInUser = () => {
        signIn('cognito', {
            callbackUrl: `${window.location.origin}/equines`
        });
    };

    const signOutUser = () => {
        signOut({
            callbackUrl: `${window.location.origin}/`
        });
    };

    return (
        <Box sx={{ flexGrow: 1 }}>
            <AppBar position="static">
                <Toolbar sx={{
                    flexDirection: "row",
                    justifyContent: "space-between"
                }}>
                    <MenuIcon sx={{ height: '2.1rem', width: '2.1rem'}}/>
                    <IconButton
                        size="large"
                        edge="start"
                        color="inherit"
                        aria-label="menu"
                        sx={{ mr: 2 }}
                    >
                        <HomeIcon />
                    </IconButton>

                    {!session && (
                        <Button color="inherit" onClick={signInUser}>
                            Login
                        </Button>
                    )}
                    {session && (
                        <Button color="inherit" onClick={signOutUser}>
                            Sign Out
                        </Button>
                    )}
                </Toolbar>
            </AppBar>
        </Box>
    );
}
