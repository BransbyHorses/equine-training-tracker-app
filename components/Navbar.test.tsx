import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import Navbar from "./Navbar";

describe("Navbar", () => {
	it("renders", () => {
		const tree = renderer.create(<Navbar />);
		expect(tree).toMatchSnapshot();
	});
});
