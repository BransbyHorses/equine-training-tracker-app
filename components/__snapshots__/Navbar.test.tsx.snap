// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders 1`] = `
<div
  className="MuiBox-root css-5liw95"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1g467ij-MuiToolbar-root"
    >
      <button
        aria-label="home"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-1e0d89p-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <div
          className="MuiBox-root css-194zdsd"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <span
            style={
              {
                "background": "none",
                "border": 0,
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": "initial",
                "left": 0,
                "margin": 0,
                "opacity": 1,
                "overflow": "hidden",
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": "initial",
              }
            }
          >
            <img
              data-nimg="fill"
              decoding="async"
              onError={[Function]}
              onLoad={[Function]}
              sizes="100vw"
              src="/assets/bransbyLogo.svg"
              srcSet="/assets/bransbyLogo.svg 640w, /assets/bransbyLogo.svg 750w, /assets/bransbyLogo.svg 828w, /assets/bransbyLogo.svg 1080w, /assets/bransbyLogo.svg 1200w, /assets/bransbyLogo.svg 1920w, /assets/bransbyLogo.svg 2048w, /assets/bransbyLogo.svg 3840w"
              style={
                {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </span>
        </div>
      </button>
      <button
        aria-label="open navigation menu"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1x8yd2s-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </button>
    </div>
  </header>
</div>
`;
